luaui/Widgets/export_defs.lua

```diff
@@ -0,0 +1,60 @@
+function widget:GetInfo()
+	return {
+		name = "Unitdefs JSON Export",
+		desc = "Exports UnitDefs into JSON files.",
+		author = "Wereii",
+		date = "June 2023",
+		license = "GNU GPL, v2 or later, BA/BAR only",
+		layer = 99999,
+		enabled = false
+	}
+end
+
+local export_folder_path = "json_export/"
+
+local function TableToFile(tbl, filename)
+	local file, err = io.open(filename, "w")
+	if not file then
+		Spring.Echo("Error opening file: " .. err)
+		return
+	end
+	file:write(Json.encode(tbl))
+	file:close()
+end
+
+local function ExportDefs()
+	if not VFS.FileExists(export_folder_path) then
+		Spring.CreateDir(export_folder_path)
+	end
+
+	for id, unitDef in pairs(UnitDefs) do
+		Spring.Echo(string.format("Exporting unitdef: %s", unitDef.name))
+		local tbl = {}
+
+		tbl["id"] = unitDef.id
+		-- embed higher-level "computed" fields like translatedHumanName, translatedTooltip, etc.
+		for field_name, value in pairs(unitDef) do
+			if type(value) ~= "table" or type(value) ~= "function" or type(value) ~= "nil" then
```

Comment:
We want to serialize functions?

Reply:
`~=` is `!=` in Lua :slightly_smiling_face:

Reply:
🤦 sorry!

Reply:
fairly sure `type(value)` can never be `nil`. on the other hand, `table` is probably fine (note that `unitDef.customParams` is already a table)

Reply:
I actually had to check the types in original documentation so, it's just a defense against receiveing nil for whatever reason (`type(nil)` returns "nil" on Lua 5.4.6, `type(type(nil)) == "string"`)

Reply:
removed the check for `nil`